python/tests/test_combinatorics.py

```diff
@@ -0,0 +1,524 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Tskit Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Test cases for combinatorial algorithms.
+"""
+import itertools
+import unittest
+
+import tskit
+import tskit.combinatorics as comb
+from tskit.combinatorics import RankTree
+
+
+def all_labelled_trees(n):
```

Comment:
Let's move these two functions into the ``combinatorics`` module and document them. They'll be really useful when explaining how this stuff works, and they're just generally cool.

I guess they should return a tskit.Tree instead of a RankTree though.

Reply:
Since we have to go through `RankTree` anyway, how about another generator function wrapper that converts `RankTree` to a tskit Tree? We can keep this name for the function that returns tskit Trees and use the other internally as this is currently being used.

Reply:
SGTM!

Reply:
Ended up exposing three functions: all trees, all tree shapes, and all labellings of a given tree